pennylane_qiskit/qiskit_device2.py

```diff
@@ -495,7 +495,9 @@ def execute_circuits(session):
                             f"Setting shot vector {circ.shots.shot_vector} is not supported for {self.name}."
                             f"The circuit will be run once with {circ.shots.total_shots} shots instead."
                         )
-                    if isinstance(circ.measurements[0], (ExpectationMP, VarianceMP)):
+                    if isinstance(circ.measurements[0], (ExpectationMP, VarianceMP)) and all(
+                        mp.obs.pauli_rep for mp in circ.measurements
+                    ):
```

Comment:
That works as well. Moving the logic into `split_measurement_types` is a good solution that ensures that measurements that can go to the Estimator do go to the estimator. I avoided this solution because (this is just personal preference) it's a bit confusing since the function name suggests that it'll split based on measurement type, but this will split it based on `measurement_type` as well as if the observable has a `pauli_rep`. But I can't think of a different way that allows Estimator-friendly measurements to actually go to the Estimator without splitting it at this place.

I think this is a good solution though, so I will implement it this way and add a comment about this edge case and why it exists.

Reply:
Makes sense as a concern - I would suggest fixing the docstring and name of the function to reflect its modified behaviour, instead of avoiding using the function. Maybe it could be renamed as `split_execution_types` or `split_primitive_types`? The docstring will probably also need an update.